packages/client/src/runtime/core/engines/common/errors/engine-not-found/nativeGeneratedOnDifferentPlatform.ts

```diff
@@ -0,0 +1,18 @@
+import { addRuntimeToYourBinaryTargets } from './addRuntimeToYourBinaryTargets'
+import { EngineNotFoundErrorInput } from './EngineNotFoundErrorInput'
+import { queryEngineCannotBeFound } from './queryEngineCannotBeFound'
+import { theseLocationsHaveBeenSearched } from './theseLocationsHaveBeenSearched'
+
+export function nativeGeneratedOnDifferentPlatform(input: EngineNotFoundErrorInput) {
+  const { runtimeBinaryTarget, generatorBinaryTargets } = input
+  const nativeBinaryTarget = generatorBinaryTargets.filter((bt) => bt.native === true)?.[0]
```

Comment:
Nit, if `bt.native` is a `boolean`, that `=== true` is superfluous

```suggestion
  const nativeBinaryTarget = generatorBinaryTargets.filter((bt) => bt.native)?.[0]
```